jsonapi/jsonapi_utility.go

```diff
@@ -167,6 +178,11 @@ func JSONErrorResponse(obj interface{}, err error) error {
 			return errs.WithStack(ctx.Conflict(jsonErr))
 		}
 	default:
+		mutex.Lock()
+		raven.CaptureErrorAndWait(err, nil)
```

Comment:
Before this we can send more info like user request object by just doing:

```go
raven.SetHttpContext(raven.NewHttp(r))
```
Here `r` is of type `http.Request`.

And also more user info like userid, email, etc. by doing:

```go
raven.SetUserContext(&raven.User{
	ID:    "123",
	Email: "suraj@example.com",
})
```

But I am not sure how do we get all that info at this point in the code?

Reply:
This is the right place to send errors to sentry. We either should send them in log.Error()/Warn()/Info() etc. or use logrus-sentry integration maybe (this needs some research).
CaptureErrorAndWait() also doesn't look right. We should not use a blocking call.

Reply:
@alexeykazakov We want to mainly collect 'unknown/unhandled' errors first, this is one of the locations. Trying to avoid flooding the server / us with pointless "errors" as I know log.Error is overused especially in WIT. far from every log.Error is something we need to do something about. log data is normally added as Breadcrumps and tracing info to the report, not be the main trigger it self.

Reply:
Yeah.. we should report true errors only. Not just info, debug etc. Agree. But if just report all errors to sentry for now then it could help us to see the trends and clean up our error reporting? For example convert some errors to info or remove duplicated errors (when we log the same error multiple times through the stack).
WDYT?

Reply:
`Log.Error(..)` is used in a lot of `resource not found` scenarios. From a WIT perspective, this isn't really an error. But from an OpenShift.io perspective, it is an error and might be of interest to understand if there's trend with which a service is sending requests to WIT which ends up in a `NotFoundError(..)` - basically for an unknown source ?